scripts/add_electricity.py

```diff
@@ -252,40 +252,65 @@ def attach_load(n, demand_profiles):
     n.madd("Load", demand_df.columns, bus=demand_df.columns, p_set=demand_df)
 
 
-def update_transmission_costs(n, costs, length_factor=1.0, simple_hvdc_costs=False):
-    n.lines["capital_cost"] = (
-        n.lines["length"] * length_factor * costs.at["HVAC overhead", "capital_cost"]
-    )
-
-    if n.links.empty:
-        return
+def attach_dc_costs(
+    n, lines_or_links, costs, length_factor=1.0, simple_hvdc_costs=False
+):
+    # TODO would it make sense to add a check on allowed keywords value?
+    if lines_or_links == "lines":
+        df_to_modify = n.lines
+    elif lines_or_links == "links":
+        df_to_modify = n.links
```

Comment:
Hello :)
Nice improvement!
However, I feel like we don't need both n and lines_or_links.
We could have lines_or_links to be the dataframe where the dc costs are modified. To do so, the definition of the function may be:
`def attach_dc_costs(lines_or_links, costs, length_factor=1.0, simple_hvdc_costs=False):`
And what you defined as df_to_modify can be replaced with lines_or_links

And when we want to attach the dc costs on lines, we could simply do:
`attach_dc_costs(n.lines, costs, length_factor, simple_hvdc_costs)`

In the case above, lines_or_links will get the value of n.lines and the code will work on lines.

By doing this, we don't need the "lines" and "links" string parameters to distinguish the two.

Under a similar principle, we could do the underwater.

Not sure if the above is clear, happy to interact if needed :)

Reply:
You comment is crystal clear and very helpful :) Implemented!